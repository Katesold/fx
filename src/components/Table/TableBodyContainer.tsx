import { TableBody, TableData, TableRow } from "./styles";
import flag from '../../assets/flag.png';
import { TableBodyContainerProps } from "./types";

export const TableBodyContainer: React.FC<TableBodyContainerProps> = ({ rows }) => {
    return (
        <TableBody>
            {rows?.map((row, index) => {
                return (
                    <TableRow key={`tableRow-${index}`}>
                        {row.map(dataCell => {
                             if(dataCell === 'GBP') {
                                return <TableData key={`tableCell-${dataCell}`}><img src={flag} alt='currency' />{dataCell}</TableData>
                            } else {
                                return <TableData key={`tableCell-${dataCell}`}>{dataCell}</TableData>
                            }
                        })}
                    </TableRow>
                )
            })}
        </TableBody>
    )
};