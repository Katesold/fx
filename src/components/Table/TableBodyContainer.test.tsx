import { TableBodyContainer } from './TableBodyContainer';
import { render } from '@testing-library/react';

describe('TableBodyContainer components are being tested', () => {
    const props = {
        rows: [
            ["6535","GBP","122","Invoice 4"],
            ["100005","GBP","255","Invoice 5"],
        ]
    };
    it('should match the snapshot', () => {
        const { container } = render(<TableBodyContainer {...props} />);
        expect(container).toMatchSnapshot();
    });
});