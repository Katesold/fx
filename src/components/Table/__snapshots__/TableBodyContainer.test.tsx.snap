// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBodyContainer components are being tested should match the snapshot 1`] = `
<div>
  <tbody
    class="sc-hLBbgP bzkRai"
  >
    <tr
      class="sc-gKPRtg fXGJDB"
    >
      <td
        class="sc-iBYQkv fcxNDm"
      >
        6535
      </td>
      <td
        class="sc-iBYQkv fcxNDm"
      >
        <img
          alt="currency"
          src="flag.png"
        />
        GBP
      </td>
      <td
        class="sc-iBYQkv fcxNDm"
      >
        122
      </td>
      <td
        class="sc-iBYQkv fcxNDm"
      >
        Invoice 4
      </td>
    </tr>
    <tr
      class="sc-gKPRtg fXGJDB"
    >
      <td
        class="sc-iBYQkv fcxNDm"
      >
        100005
      </td>
      <td
        class="sc-iBYQkv fcxNDm"
      >
        <img
          alt="currency"
          src="flag.png"
        />
        GBP
      </td>
      <td
        class="sc-iBYQkv fcxNDm"
      >
        255
      </td>
      <td
        class="sc-iBYQkv fcxNDm"
      >
        Invoice 5
      </td>
    </tr>
  </tbody>
</div>
`;
